server_http.hpp

```diff
@@ -80,6 +80,7 @@ namespace SimpleWeb {
             std::string remote_endpoint_address;
             unsigned short remote_endpoint_port;
             
+            std::function<void(const std::exception&)> exception_handler;
```

Comment:
Why do we need the read_remote_endpoint_data?
```cpp
diff --git a/server_http.hpp b/server_http.hpp
index 70a86a2..440d36f 100644
--- a/server_http.hpp
+++ b/server_http.hpp
@@ -80,22 +80,10 @@ namespace SimpleWeb {
             std::string remote_endpoint_address;
             unsigned short remote_endpoint_port;
             
-            std::function<void(const std::exception&)> exception_handler;
         private:
             Request(): content(streambuf) {}
             
             boost::asio::streambuf streambuf;
-            
-            void read_remote_endpoint_data(socket_type& socket) {
-                try {
-                    remote_endpoint_address=socket.lowest_layer().remote_endpoint().address().to_string();
-                    remote_endpoint_port=socket.lowest_layer().remote_endpoint().port();
-                }
-                catch(const std::exception&e) {
-                    if(exception_handler)
-                       exception_handler(e);
-                }
-            }
         };
         
         class Config {
@@ -223,8 +211,14 @@ namespace SimpleWeb {
             //Create new streambuf (Request::streambuf) for async_read_until()
             //shared_ptr is used to pass temporary objects to the asynchronous functions
             std::shared_ptr<Request> request(new Request());
-            request->read_remote_endpoint_data(*socket);
 
+            try {
+                request->remote_endpoint_address=socket->lowest_layer().remote_endpoint().address().to_string();
+                request->remote_endpoint_port=socket->lowest_layer().remote_endpoint().port();
+            } catch(const std::exception &e){
+                if(exception_handler)
+                    exception_handler(e);
+            }
             //Set timeout on the following boost::asio::async-read or write function
             std::shared_ptr<boost::asio::deadline_timer> timer;
             if(timeout_request>0)
```
Is it to hide `boost::asio`  to implementors of the server?